packages/gatsby/src/utils/detect-port-in-use-and-prompt.js

```diff
@@ -0,0 +1,30 @@
+const detect = require(`detect-port`)
+const report = require(`gatsby-cli/lib/reporter`)
+
+// Checks if a port is in use and prompts the user to enter another one
+// Then calls callback with new port
+const detectPortInUseAndPrompt = (port, rlInterface, callback) => {
+  let newPort = port
+
+  detect(port, (err, _port) => {
```

Comment:
how about making a promise interface?
```
return new Promise((resolve, reject) =>
  detect(port, (err, __port) => {
    ...
    resolve(port);
  })
);
```

Reply:
The function is sync, I just needed to conditionally call a passed in function (the callback) 🤔

Reply:
```js
rlInterface.question(question, answer => {
        if (answer.length === 0 || answer.match(/^yes|y$/i)) {
          newPort = _port
        }
        callback(newPort)
      })
```
this part is async as we're waiting for an answer, we block everything else but technically it's async but it's not a big deal I just like promise syntax better than callbacks especially in `develop.js`

Reply:
I'm fine merging this--but it does show a promise based interface (they show it with a `yield`) in the documentation.